src/terminal/adapter/adaptDispatchGraphics.cpp

```diff
@@ -83,24 +100,36 @@ void AdaptDispatch::_SetRgbColorsHelperFromSubParams(const VTParameter colorItem
                                                      const VTSubParameters options,
                                                      TextAttribute& attr) noexcept
 {
-    // This should be called for applying FG and BG colors only.
-    assert(colorItem == GraphicsOptions::ForegroundExtended ||
-           colorItem == GraphicsOptions::BackgroundExtended);
+    const auto applyColor = [&](const TextColor& color) {
+        switch (colorItem)
+        {
+        case ForegroundExtended:
+            attr.SetForeground(color);
+            break;
+        case BackgroundExtended:
+            attr.SetBackground(color);
+            break;
+        case UnderlineColor:
+            attr.SetUnderlineColor(color);
+            break;
+        default:
+            break;
+        };
+    };
 
-    const bool isForeground = (colorItem == GraphicsOptions::ForegroundExtended);
     const DispatchTypes::GraphicsOptions typeOpt = options.at(0);
-
-    if (typeOpt == DispatchTypes::GraphicsOptions::RGBColorOrFaint)
+    switch (typeOpt)
+    {
+    case DispatchTypes::GraphicsOptions::RGBColorOrFaint:
     {
         // sub params are in the order:
         // :2:<color-space-id>:<r>:<g>:<b>
 
-        // We treat a color as invalid, if it has a color space ID, as some
-        // applications that support non-standard ODA color sequence may send
-        // the red value in its place.
+        // We treat a color as invalid if it has a non-empty color space ID, as
+        // some applications that support non-standard ODA color sequence might
+        // send the red value in its place.
         const bool hasColorSpaceId = options.at(1).has_value();
 
-        // Skip color-space-id.

```

Comment:
I think this is not needed, given that we are reading color-space-id, which wasn't the case when I added this comment😅